### PR TITLE
ci: some minor fixes

### PR DESCRIPTION
**Pull request by henryiii**
Fixes to get CI working again, and some speed improvements.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,14 +1,22 @@
 name: CI
-on: [push, pull_request]
+on:
+  push:
+    branches: [main]
```
Comment (henryiii) [inline .github/workflows/ci.yml:4]: Without this, CI is currently running twice on things like dependabot PRs.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+  pull_request:
+
 env:
   FORCE_COLOR: "1"
-  PRE_COMMIT_COLOR: "always"
```
Comment (henryiii) [inline .github/workflows/ci.yml:5]: I believe this was fixed in pre-commit.

Comment (henryiii) [reply .github/workflows/ci.yml:5]: Hmm, maybe not. Just a sec.

Comment (henryiii) [reply .github/workflows/ci.yml:5]: No, I take that back.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -18,10 +26,11 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Set up Miniconda
         uses: conda-incubator/setup-miniconda@v2
-        if: matrix.python-version != '3.11'
         with:
           auto-update-conda: true
           python-version: ${{ matrix.python-version }}
+          miniforge-variant: Mambaforge
+          use-mamba: true
```
Comment (henryiii) [inline .github/workflows/ci.yml:33]: A lot faster & supports 3.11.

**Comment by henryiii** [comment]
The coverage issue:

https://github.com/wntrblm/nox/blob/cc710bde9d6a8781833144bac02a5f4581d9eca7/nox/virtualenv.py#L373-L394

I think this is never None on Ubuntu 22.04 & Python 3.11. Shouldn't the 100% requirement be on the combined coverage, rather than partial? Anyway, should this be marked as not being required to be covered, etc?

**Review by crwilcox — APPROVED** [review]

**Comment by FollowTheProcess** [comment]
Hmmm wonder why it would never be `None` on ubuntu-latest py3.11? Weird.

Agree on combined coverage being used for the `fail-under`, is this a config option in coverage? I did have a quick look but couldn't see anything obvious.

If we can't do that then yeah I agree we should exclude it from coverage measurement.

This all looks good to me 👍🏻

**Pull request merged**